### PR TITLE
Adds `delete` operation for non-contained navigation properties only if explicitly allowed via annotation

### PR DESCRIPTION
**Pull request by irvinesunday**
Fixes https://github.com/microsoft/OpenAPI.NET.OData/issues/483

This PR:
- Ensures `delete` operations for non-contained nav. props is generated only if `DeleteRestrictions` exists for this nav. prop and the `Deletable` property is set to `true`.
- Updates test to validate the above.
- Updates release note.
- Minor documentation text fixes to the convert settings.

### CODE REVIEW

File: src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs [hunk 1]
```diff
@@ -197,27 +197,28 @@ private void AddDeleteOperation(OpenApiPathItem item, NavigationPropertyRestrict
...
+                    AddOperation(item, OperationType.Delete);
+                }
             }
             else
             {
-                if ((navPropDeleteRestriction?.IsDeletable ?? false) &&
-                    (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
+                // Add delete operation for non-contained nav. props only if explicitly set to true via annotation
+                // Note: Use Deletable and not IsDeletable
+                if ((navPropDeleteRestriction?.Deletable ?? false) &&
```
Comment (baywet) [inline src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs:220]: collapse this condition into the parent else

**Review by baywet — CHANGES_REQUESTED** [review]

File: src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs (hunk 1 above)
Comment (irvinesunday) [reply src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs]: Resolved

**Review by andrueastman — APPROVED** [review]

File: src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs [hunk 2]
```diff
@@ -197,31 +197,29 @@ private void AddDeleteOperation(OpenApiPathItem item, NavigationPropertyRestrict
...
-                {
-                    if (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many || LastSegmentIsKeySegment)
-                    {
-                        AddOperation(item, OperationType.Delete);
-                    }
-                }
+                DeleteRestrictionsType entityDeleteRestriction = Context.Model.GetRecord<DeleteRestrictionsType>(_navPropEntityType);
+                bool isDeletableDefault = navPropDeleteRestriction == null && entityDeleteRestriction == null;
+
+                if ((isDeletableDefault ||
```
Comment (baywet) [inline src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs:207]: collapse this if into the parent statement

File: src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs [hunk 3]
```diff
@@ -197,31 +197,29 @@ private void AddDeleteOperation(OpenApiPathItem item, NavigationPropertyRestrict
...
-            {
-                if ((navPropDeleteRestriction?.IsDeletable ?? false) &&
-                    (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
-                    LastSegmentIsKeySegment))
-                {
-                    AddOperation(item, OperationType.Delete);
-                }
+            else if ((navPropDeleteRestriction?.Deletable ?? false) &&
+                    (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
+                    LastSegmentIsKeySegment))
```
Comment (baywet) [inline src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs:218]: this seems to be a requirement for both blocks, what about extracting that statement, negating it and returning early before those two blocks to improve readability?

**Review by baywet — CHANGES_REQUESTED** [review]

File: src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs (hunk 2 above)
Comment (irvinesunday) [reply src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs]: Resolved

File: src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs (hunk 3 above)
Comment (irvinesunday) [reply src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs]: Resolved

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**